duo-universal-sdk/src/main/java/com/duosecurity/service/DuoConnector.java

```diff
@@ -22,24 +25,35 @@ public class DuoConnector {
   /**
    * DuoConnector Constructor.
    *
-   * @param apiHost         This value is the api host provided by Duo in the admin panel.
-   * @param caCerts         CA Certificates used to connect to Duo
+   * @param apiHost This value is the api host provided by Duo in the admin panel.
+   * @param caCerts CA Certificates used to connect to Duo
    *
-   * @throws DuoException   For issues getting and validating the URL
+   * @throws DuoException For issues getting and validating the URL
    */
   public DuoConnector(String apiHost, String[] caCerts) throws DuoException {
-    CertificatePinner duoCertificatePinner = new CertificatePinner.Builder()
-        .add(apiHost, caCerts)
-        .build();
+    this(apiHost, null, null, caCerts);
+  }
 
-    OkHttpClient client = new OkHttpClient.Builder().certificatePinner(
-                            duoCertificatePinner).build();
+  /**
+   * DuoConnector Constructor.
+   *
+   * @param apiHost This value is the api host provided by Duo in the admin panel.
+   * @param caCerts CA Certificates used to connect to Duo
+   *
+   * @throws DuoException For issues getting and validating the URL
+   */
+  public DuoConnector(String apiHost, String proxyHost, Integer proxyPort, String[] caCerts) throws DuoException {
+    CertificatePinner duoCertificatePinner = new CertificatePinner.Builder().add(apiHost, caCerts).build();
+    OkHttpClient client;
+    if (proxyHost != null && proxyPort != null) {
+      Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(proxyHost, proxyPort));
+      client = new OkHttpClient.Builder().certificatePinner(duoCertificatePinner).proxy(proxy).build();
+    } else {
+      client = new OkHttpClient.Builder().certificatePinner(duoCertificatePinner).build();
+    }
 
-    retrofit = new Retrofit.Builder()
-            .baseUrl(getAndValidateUrl(apiHost, "").toString())
-            .addConverterFactory(JacksonConverterFactory.create())
-            .client(client)
-            .build();
+    retrofit = new Retrofit.Builder().baseUrl(getAndValidateUrl(apiHost, "").toString())
```

Comment:
small nit, I think the old code is slightly cleaner to read as how retrofit being built. Just personal preference.

Reply:
@yizshi When I look at the full file it wraps (.baseUrl starts on the next line after Builder(), as in the original, but perhaps my viewport size is doing that?